user_interface/EbikeUIController/EbikeUIController.ino

```diff
@@ -0,0 +1,720 @@
+	0x00, 0x00, 0x3c, 0xf0, 0x00, 0x00, 0x3c, 0xf0, 0x00, 0x00, 0x3c, 0xf0, 0x00, 0x00, 0x3c, 0xf0, 
+	0x00, 0x0f, 0x3c, 0xf0, 0x00, 0x0f, 0x3c, 0xf0, 0x00, 0x0f, 0x3c, 0xf0, 0x00, 0x0f, 0x3c, 0xf0, 
+	0x03, 0xcf, 0x3c, 0xf0, 0x03, 0xcf, 0x3c, 0xf0, 0x03, 0xcf, 0x3c, 0xf0, 0x03, 0xcf, 0x3c, 0xf0, 
+	0xf3, 0xcf, 0x3c, 0xf0, 0xf3, 0xcf, 0x3c, 0xf0, 0xf3, 0xcf, 0x3c, 0xf0, 0xf3, 0xcf, 0x3c, 0xf0
+};
+// Array of all bitmaps for convenience. (Total bytes used to store images in PROGMEM = 624)
+// const int epd_bitmap_allArray_LEN = 6;
+// const unsigned char* epd_bitmap_allArray[6] = {
+// 	epd_bitmap_Battery_Level_Sym_0_,
+// 	epd_bitmap_Battery_Level_Sym_100_,
+// 	epd_bitmap_Battery_Level_Sym_20_,
+// 	epd_bitmap_Battery_Level_Sym_40_,
+// 	epd_bitmap_Battery_Level_Sym_60_,
+// 	epd_bitmap_Battery_Level_Sym_80_
+// };
+
+// Global variables to store different parameters
+int speed = 0;                    // Speed of the e-bike
+int throttle = 0;                 // Max throttle percentage
+int power = 0;                    // E-bike power output OR regenerative power
+int temp = 0;                     // Temperature of the e-bike system
+int batteryRange = 0;             // Battery range of the e-bike
+int regenMethod = 1;              // Regenerative braking method (1 or 2)
+
+// Variables to convert system parameters from int to display format
+char speed_string[10];
+int speed_string_length;
+
+char throttle_string[10];
+int throttle_string_length;
+
+char power_string[10];
+int power_string_length;
+
+char temp_string[10];
+int temp_string_length;
+
+char battery_string[10];
+int battery_string_length;
+
+int activeRegen = 1;
+
+// Create Toggle instances for buttons to increase throttle, decrease throttle, and toggle regenerative braking
+Toggle increaseThrottle(6);       // Button to increase throttle
+Toggle decreaseThrottle(7);       // Button to decrease throttle
+Toggle toggleRegen(8);            // Button to toggle regenerative braking method
+
+void setup() {
+
+  // Initialize the OLED display
+  u8g2.begin();
+
+  // Display the logo during setup
+  u8g2.firstPage();
+  do {
+    // Draw logo during system startup
+    u8g2.drawBitmap(0, 0, 128/8, 64, epd_bitmap_REBIKE_Logo);
+  } while (u8g2.nextPage());
+
+  // Wait for 4 seconds to show the logo
+  delay(4000);
+
+   // Initialize the button toggles
+  increaseThrottle.begin(0);
+  decreaseThrottle.begin(1);
+  toggleRegen.begin(2);
+
+  // This pin is connected to a potentiometer for testing purposes.
+  pinMode(14, INPUT);
+
+  }
+
+void loop() {
+
+  // Poll the state of each button
+  increaseThrottle.poll();
+  decreaseThrottle.poll();
+  toggleRegen.poll();
+
+  // Functions to read incoming signals from the motor controller and BMS to update various
+  // system parameters for UI display (speed, power, temperature, battery range)
+  updateSpeed();
+  updatePower();
+  updateTemp();
+  updateBatteryRange();
```

Comment:
A little confused why these function calls compile, typically functions have to be declared somewhere before they get called. A highly reputable source (some random dude on a forum) says that the Arduino IDE will try to auto generate these declarations for you, but it is not always correct and can result in difficult to track errors. To be safe, I think all the functions you defined below loop() should have declarations at the top of the file, above the global variables.

Reply:
This modification has been implemented as per your request in most recent commit.